src/main/java/com/google/api/gax/grpc/OperationCallable.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.google.longrunning.GetOperationRequest;
+import com.google.longrunning.Operation;
+import com.google.longrunning.OperationsApi;
+import com.google.protobuf.Message;
+import io.grpc.Channel;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * An OperationCallable is an immutable object which is capable of initiating RPC calls to
+ * long-running API methods and returning an OperationFuture to manage the polling of the
+ * Operation and getting the response.
+ */
+public final class OperationCallable<RequestT, ResponseT extends Message> {
+  private final UnaryCallable<RequestT, Operation> initialCallable;
+  private final Channel channel;
+  private final ScheduledExecutorService executor;
+  private final OperationsApi operationsApi;
+  private final Class<ResponseT> responseClass;
+  private final OperationCallSettings settings;
+
+  /** Package-private for internal use. */
+  OperationCallable(
+      UnaryCallable<RequestT, Operation> initialCallable,
+      Channel channel,
+      ScheduledExecutorService executor,
+      OperationsApi operationsApi,
+      Class<ResponseT> responseClass,
+      OperationCallSettings settings) {
+    this.initialCallable = Preconditions.checkNotNull(initialCallable);
+    this.channel = channel;
+    this.executor = executor;
+    this.operationsApi = operationsApi;
+    this.responseClass = responseClass;
+    this.settings = settings;
+  }
+
+  /**
+   * Create an OperationCallable with a bound channel. If a call is made without specifying a
+   * channel, the {@code boundChannel} is used instead.
+   *
+   * <p>Package-private for internal use.
+   */
+  OperationCallable<RequestT, ResponseT> bind(Channel boundChannel) {
+    return new OperationCallable<>(
+        initialCallable, boundChannel, executor, operationsApi, responseClass, settings);
+  }
+
+  /**
+   * Create a callable object that represents a long-running operation. Public only for technical
```

Comment:
Nit: There is some inconsistency here, we use "Creates" in some of the other places.